tsconfig.json

```diff
@@ -2,7 +2,8 @@
   "compilerOptions": {
     "target": "es2020",
     "lib": ["ES2021", "DOM"],
-    "module": "CommonJS",
+    "module": "NodeNext",
+    "moduleResolution": "NodeNext",
```

Comment:
There are some considerable implications in changing the module format. I think this is something we should do, but we'll want to monitor for impact + maybe do a major version bump